src/JUnit.Xml.TestLogger/JunitXmlSerializer.cs

```diff
@@ -308,6 +307,34 @@ private XElement CreateTestCaseElement(TestResultInfo result)
                 testcaseElement.Add(skippedElement);
             }
 
+            StringBuilder stdErr = new StringBuilder();
+            foreach (var m in result.Messages)
+            {
+                if (TestResultMessage.StandardErrorCategory.Equals(m.Category, StringComparison.OrdinalIgnoreCase))
+                {
+                    stdErr.AppendLine(m.Text);
+                }
+            }
+
+            if (!string.IsNullOrWhiteSpace(stdErr.ToString()))
+            {
+                testcaseElement.Add(new XElement("system-err", stdErr.ToString()));
+            }
+
+            StringBuilder stdOut = new StringBuilder();
+            foreach (var m in result.Messages)
+            {
+                if (TestResultMessage.StandardOutCategory.Equals(m.Category, StringComparison.OrdinalIgnoreCase))
+                {
+                    stdOut.AppendLine(m.Text);
+                }
+            }
+
+            if (!string.IsNullOrWhiteSpace(stdOut.ToString()))
+            {
+                testcaseElement.Add(new XElement("system-out", stdOut.ToString()));
```

Comment:
I also didn't wrap this with `XCData` like the NUnit one because I wanted to match the existing system-out/system-err printing. I don't know enough about JUnit to know if parsers should expect CDATA or not.

Reply:
Also something we could do a quick test on. But if the logged text has any xml characters that need to be escaped, I expect thats what will happen here. If we wrap it in CDATA, the characters won't be escaped and should therefore be more readable. 

https://en.wikipedia.org/wiki/CDATA

Reply:
@Siphonophora would you like me to wrap the per-testcase output in CDATA and the per-testsuite output? I'd consider this to be a breaking change because it's not currently wrapped on line 229.

Reply:
I'm not sure I would call it a breaking change so long as we are still passing the xml validation. But, we can break this out to its own issue and decide on it later.